native/libcst/src/parser/grammar.rs

```diff
@@ -0,0 +1,2930 @@
+                make_if(i, a, col, b, Some(OrElse::Elif(elif)), true)
+            }
+            / i:lit("elif") a:named_expression() col:lit(":") b:block() el:else_block()? {
+                make_if(i, a, col, b, el.map(OrElse::Else), true)
+            }
+
+        rule else_block() -> Else<'a>
+            = el:lit("else") col:lit(":") b:block() {
+                make_else(el, col, b)
+            }
+
+        rule while_stmt() -> While<'a>
+            = kw:lit("while") test:named_expression() col:lit(":") b:block() el:else_block()? {
+                make_while(kw, test, col, b, el)
+            }
+
+        rule for_stmt() -> For<'a>
+            = f:lit("for") t:star_targets() i:lit("in") it:star_expressions()
+                c:lit(":") b:block() el:else_block()? {
+                    make_for(None, f, t, i, it, c, b, el)
+            }
+            / asy:tok(Async, "ASYNC") f:lit("for") t:star_targets() i:lit("in")
+                it:star_expressions()
+                c:lit(":") b:block() el:else_block()? {
+                    make_for(Some(asy), f, t, i, it, c, b, el)
+            }
+
+        rule with_stmt() -> With<'a>
+            = kw:lit("with") items:separated(<with_item()>, <comma()>)
+                col:lit(":") b:block() {
+                    make_with(None, kw, comma_separate(items.0, items.1, None), col, b)
+            }
+            / asy:tok(Async, "ASYNC") kw:lit("with") items:separated(<with_item()>, <comma()>)
+                col:lit(":") b:block() {
+                    make_with(Some(asy), kw, comma_separate(items.0, items.1, None), col, b)
+            }
+
+        rule with_item() -> WithItem<'a>
+            = e:expression() a:lit("as") t:star_target() &(lit(",") / lit(":")) {
+                make_with_item(e, Some(a), Some(t))
+            }
+            / e:expression() {
+                make_with_item(e, None, None)
+            }
+
+        rule named_expression() -> Expression<'a>
+            = a:name() op:lit(":=") b:expression() {
+                Expression::NamedExpr(make_named_expr(a, op, b))
+            }
+            / e:expression() !lit(":=") { e }
+
+        rule annotated_rhs() -> Expression<'a>
+            = yield_expr() / star_expressions()
+
+        rule global_stmt() -> Global<'a>
+            = kw:lit("global") init:(n:name() c:comma() {(n, c)})* last:name() {
+                make_global(kw, init, last)
+            }
+
+        rule nonlocal_stmt() -> Nonlocal<'a>
+            = kw:lit("nonlocal") init:(n:name() c:comma() {(n, c)})* last:name() {
+                make_nonlocal(kw, init, last)
+            }
+
+        rule yield_stmt() -> Expression<'a>
+            = yield_expr()
+
+        rule assert_stmt() -> Assert<'a>
+            = kw:lit("assert") test:expression() rest:(c:comma() msg:expression() {(c, msg)})? {
+                make_assert(kw, test, rest)
+            }
+
+        rule del_stmt() -> Del<'a>
```

Comment:
[nit / curiosity] this diverges from the grammar because we have separate rules for del_target and del_targets

Is that a historical thing, e.g. we wrap one in a tuple and not the other so it's easier to have two handler actions?

Reply:
Yeah this change is because I needed to be 100% backwards compatible with the pure python parser, which produces a tuple in one case and not the other.